Fix contradictory statements in XSS doc (#35680)
dotnet/AspNetCore.Docs: aspnetcore/security/cross-site-scripting.md

At a basic level, XSS works by tricking your application into inserting a `<script>` tag into your rendered page, or by inserting an `On*` event into an element. Developers should use the following prevention steps to avoid introducing XSS into their applications:

1. Never put untrusted data into your HTML input, unless you follow the rest of the steps below. Untrusted data is any data that may be controlled by a cyberattacker, such as HTML form inputs, query strings, HTTP headers, or even data sourced from a database, as a cyberattacker may be able to breach your database even if they can't breach your application.

> 1. Before putting untrusted data into an HTML attribute, ensure that it's HTML attribute-encoded. This specialized form of HTML encoding handles double quotes ("), single quotes ('), ampersands (&), and less-than (<) characters. when dealing with untrusted input,Use HTML encoding for general HTML content and HTML attribute encoding specifically for HTML attributes.

Reviewer: ```suggestion
1. Before putting untrusted data into an HTML attribute, ensure that it's HTML attribute-encoded. This specialized form of HTML encoding handles double quotes ("), single quotes ('), ampersands (&), and less-than (<) characters. When dealing with untrusted input, use HTML encoding for general HTML content and HTML attribute encoding specifically for HTML attributes.
```

Reviewer: Just some minor typos fixed.

At a basic level, XSS works by tricking your application into inserting a `<script>` tag into your rendered page, or by inserting an `On*` event into an element. Developers should use the following prevention steps to avoid introducing XSS into their applications:

1. Never put untrusted data into your HTML input, unless you follow the rest of the steps below. Untrusted data is any data that may be controlled by a cyberattacker, such as HTML form inputs, query strings, HTTP headers, or even data sourced from a database, as a cyberattacker may be able to breach your database even if they can't breach your application.

> 1. Before putting untrusted data into an HTML element, ensure thattit's HTML encoded. HTML encoding takes characters such as &lt; and changes them into a safe form like &amp;lt;.

Reviewer: ```suggestion
1. Before putting untrusted data into an HTML element, ensure that it's HTML encoded. HTML encoding takes characters such as &lt; and changes them into a safe form like &amp;lt;.
```

> 1. Before putting untrusted data into an HTML attribute, ensure that it's HTML attribute encoded. This specialized form of HTML encoding handles double quotes (\"), single quotes ('\), ampersands (\&), and less-than (\<) characters. when dealing with untrusted input,Use HTML encoding for general HTML content and HTML attribute encoding sfor HTML attributes.

Reviewer: ```suggestion
1. Before putting untrusted data into an HTML attribute, ensure that it's HTML attribute encoded. This specialized form of HTML encoding handles double quotes (\"), single quotes ('\), ampersands (\&), and less-than (\<) characters. When dealing with untrusted input, use HTML encoding for general HTML content and HTML attribute encoding for HTML attributes.
```

Adding in again for typo fix.  I think we wrote at the same tim.

> 1. Before putting untrusted data into an HTML attribute, ensure that it's HTML attribute encoded. This specialized form of HTML encoding handles double quotes (\"), single quotes ('\), ampersands (\&), and less-than (\<) characters. when dealing with untrusted input, Use HTML encoding for general HTML content and HTML attribute encoding for HTML attributes.

Reviewer: ```suggestion
1. Before putting untrusted data into an HTML attribute, ensure that it's HTML attribute encoded. This specialized form of HTML encoding handles double quotes (\"), single quotes ('\), ampersands (\&), and less-than (\<) characters. When dealing with untrusted input, use HTML encoding for general HTML content and HTML attribute encoding for HTML attributes.
```

One more try to sync.  haha.
